arm/Microsoft.Network/azureFirewalls/readme.md

```diff
@@ -42,16 +46,76 @@ This module deploys a firewall.
 | `diagnosticWorkspaceId` | string | `''` |  | Log Analytics workspace resource identifier. |
 | `enableDefaultTelemetry` | bool | `True` |  | Enable telemetry via the Customer Usage Attribution ID (GUID). |
 | `firewallPolicyId` | string | `''` |  | Resource ID of the Firewall Policy that should be attached. |
+| `isCreateDefaultPublicIP` | bool | `True` |  | Specifies if a public ip should be created by default if one is not provided. |
 | `location` | string | `[resourceGroup().location]` |  | Location for all resources. |
 | `lock` | string | `'NotSpecified'` | `[CanNotDelete, NotSpecified, ReadOnly]` | Specify the type of lock. |
 | `natRuleCollections` | array | `[]` |  | Collection of NAT rule collections used by Azure Firewall. |
 | `networkRuleCollections` | array | `[]` |  | Collection of network rule collections used by Azure Firewall. |
+| `publicIPAddressObject` | object | `{object}` |  | Specifies the properties of the public IP to create and be used by Azure Firewall. If it's not provided and publicIPAddressId is empty, a '-pip' suffix will be appended to the Firewall's name. |
 | `roleAssignments` | array | `[]` |  | Array of role assignment objects that contain the 'roleDefinitionIdOrName' and 'principalId' to define RBAC role assignments on this resource. In the roleDefinitionIdOrName attribute, you can provide either the display name of the role definition, or its fully qualified ID in the following format: '/providers/Microsoft.Authorization/roleDefinitions/c2f4ef07-c644-48eb-af81-4b1b4947fb11'. |
 | `tags` | object | `{object}` |  | Tags of the Azure Firewall resource. |
 | `threatIntelMode` | string | `'Deny'` | `[Alert, Deny, Off]` | The operation mode for Threat Intel. |
 | `zones` | array | `[1, 2, 3]` |  | Zone numbers e.g. 1,2,3. |
 
 
+### Parameter Usage: `additionalPublicIpConfigurations`
```

Comment:
A new PR was just merged that adds some updates to the ReadMEs.

For this purpose please
- Merge latest main
- Re-run the ReadMe script
- Updated the new `'Parameter Usage'` blocks with the new structure, that is both a Bicep & Json block. To make it easier for you, here are the blocks for bicep


```Bicep
additionalPublicIpConfigurations: {
    value: [
        {
            name: 'ipConfig01'
            publicIPAddressResourceId: '/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/publicIPAddresses/adp-<<namePrefix>>-az-pip-x-fw-01'
        }
        {
            name: 'ipConfig02'
            publicIPAddressResourceId: '/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/publicIPAddresses/adp-<<namePrefix>>-az-pip-x-fw-02'
        }
    ]
}
```

```Bicep
publicIPAddressObject: {
    value: {
        name: 'mypip'
        publicIPPrefixResourceId: '/subscriptions/<<subscriptionId>>/resourceGroups/validation-rg/providers/Microsoft.Network/publicIPPrefixes/myprefix'
        publicIPAllocationMethod: 'Dynamic'
        skuName: 'Basic'
        skuTier: 'Regional'
        roleAssignments: {
            value: [
                {
                    roleDefinitionIdOrName: 'Reader'
                    principalIds: [
                        '<<deploymentSpId>>'
                    ]
                }
            ]
        }
        diagnosticMetricsToEnable: {
            value: [
                'AllMetrics'
            ]
        }
        diagnosticLogCategoriesToEnable: {
            value: [
                'DDoSProtectionNotifications'
                'DDoSMitigationFlowLogs'
                'DDoSMitigationReports'
            ]
        }
    }
}
```